File: python_cpp/uPIMulator_frontend/src/abi/isa/cc/const_cc_zero.py
from typing import Set

from abi.isa.cc._base_cc import BaseCC
from abi.isa.instruction.condition import Condition


class ConstCCZero(BaseCC):
    def __init__(self, condition: Condition):
        super().__init__(ConstCCZero.conditions(), condition)

    @staticmethod
    def conditions() -> Set[Condition]:
        return {Condition.Z}

File: golang_vm/uPIMulator/src/device/linker/kernel/instruction/cc/const_cc_zero.go
package cc

import (
	"errors"
)

type ConstCcZero struct {
	condition Condition
}

func (this *ConstCcZero) Init(condition Condition) {
	conditions := map[Condition]bool{
		Z: true,
	}

	if _, found := conditions[condition]; !found {
		err := errors.New("condition is not allowed")
		panic(err)
	}

	this.condition = condition
}

func (this *ConstCcZero) Condition() Condition {
	return this.condition
}
